.github/workflows/upload-nightly-release.yml

```
@@ -0,0 +1,47 @@
+name: Build nightly PennyLane releases for TestPyPI
+
+on:
+  schedule:
+    # Run every weekday at 23:40 EDT (cron is in UTC)
+    - cron: "40 3 * * 2-6"
+  workflow_dispatch:
+
+jobs:
+  setup:
+    name: Setup the release
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout PennyLane repo
+      uses: actions/checkout@v4
+      with:
+        ssh-key: ${{ secrets.NIGHTLY_VERSION_UPDATE_DEPLOY_KEY }}
+
+    - name: Bump dev version
+      run: |
+        python $GITHUB_WORKSPACE/.github/workflows/scripts/set_nightly_version.py
+
+    - name: Push new version
+      run: |
+        git config --global user.email '${{ secrets.AUTO_UPDATE_VERSION_RINGO_EMAIL }}'
+        git config --global user.name "ringo-but-quantum"
+        git add $GITHUB_WORKSPACE/pennylane/_version.py
+        git commit -m "[no ci] bump nightly version"
+        git push
+
+  upload:
+    name: Upload wheels to TestPyPI
+    needs: [setup]
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write  # IMPORTANT: mandatory for trusted publishing
+
+    steps:
+    - name: Build wheels
+      run: |
+        python setup.py bdist_wheel --dist-dir dist
```

```suggestion
        python -m pip install build
        python -m build
```

The setup.py version is no longer allowed with the recent PEP changes. It is recommended to use a builder as above (https://packaging.python.org/en/latest/tutorials/packaging-projects/).

---

True we don't do it this way, I just copied was in the existing PL wheel action. Happy to change it :)

---

Quick question, do you know how `build` compares to `pip wheel` (which is what we use for Catalyst)?

---

Build is just a frontend (https://github.com/pypa/build). It will call wheel under the hood, after it sets up an isolated environment to do the packaging.

It makes use of the information in the pyproject.toml directly for any/all needs (which is bare for PL, but will likely be updated later as needed).

---

One difference I can see to the `bdist_wheel` command is that your suggestion builds both the wheel and the source dist. I'm thinking for the nightlies this is not needed, would you be okay with the following?
```suggestion
        python -m pip install build
        python -m build --wheel --outdir dist
```